src/components/PeopleFilters.tsx

```diff
@@ -1,12 +1,58 @@
+import { useSearchParams } from 'react-router-dom';
+import classNames from 'classnames';
+import { SearchParams, getSearchWith } from '../utils/searchHelper';
+import { PersonSex } from '../types/PersonSex';
+import {
+  SEX_FEMALE, SEX_MALE, centuriesArray, initialParams,
+} from '../utils/constants';
+import { SearchLink } from './SearchLink';
+
 export const PeopleFilters = () => {
+  const [searchParams, setSearchParams] = useSearchParams();
+
+  const query = searchParams.get('query') || '';
+  const centuries = searchParams.getAll('centuries') || [];
+  const sex = searchParams.get('sex') || null;
+
+  function setSearchWith(paramsToUpdate: SearchParams) {
+    const search = getSearchWith(searchParams, paramsToUpdate);
+
+    setSearchParams(search);
+  }
+
+  const getCenturies = (century: string) => {
+    return centuries.includes(century)
+      ? centuries.filter(number => number !== century)
+      : [...centuries, century];
+  };
+
+  function handleQueryChange(event: React.ChangeEvent<HTMLInputElement>) {
+    setSearchWith({ query: event.target.value || null });
+  }
```

Comment:
Let's remove `setSearchWith` function

```suggestion
  function handleQueryChange(event: React.ChangeEvent<HTMLInputElement>) {
    const paramsToUpdate = { query: event.target.value || null }
    
    const search = getSearchWith(searchParams, paramsToUpdate);

    setSearchParams(search)
  }
```